todo_app/view_model.py

```diff
@@ -0,0 +1,56 @@
+from datetime import datetime, timedelta, time
+
+# Decorator for get_cards function to slice result by time.
+def split_result_by_updated_time(func):
```

Comment:
This is a really interesting way of solving this issue, but it looks pretty complex!

Note that a slightly different choice of default argument could make this much simpler - by switching from None defaults to `datetime.min` & `datetime.max` respectively, we can always assume we use the last of the checks
```python
def split_result_by_updated_time(func):
    def wrapper_split_result_by_updated_time(self, list_category, \
        from_threshold = datetime.min, to_threshold = datetime.max):
        intermediate_list = func(self,list_category)
        final_list = []
        for x in intermediate_list:
            if from_threshold <= x.updated_time < to_threshold: 
                final_list.append(x)
        return final_list
    return wrapper_split_result_by_updated_time
```

Reply:
datetime.min and datetime.max were good replacements for None, and I was able to eliminate two checks 👍 .. However, I retained one of the conditions as a check to ensure function doesn't unnecessarily iterate over all the cards/items when it doesn't need to. Thanks !